projects/canopy/src/lib/skeleton/skeleton.directive.ts

```diff
@@ -0,0 +1,19 @@
+import { Directive, ElementRef, Renderer2, Input, HostBinding } from '@angular/core';
+
+@Directive({
+  selector: '[lgSkeleton]',
+})
+export class LgSkeletonDirective {
+  @Input('lgSkeleton') lgSkeletonWidth = 4;
+
+  @HostBinding('style.width')
+  get backgroundSize(): string {
+    return `${this.lgSkeletonWidth}em`;
```

Comment:
The issue I'm seeing with this is that you cannot have it added conditionally e.g. `[lgSkeleton]="loading ? '8' : null"`. You could check for `null` values here though.

Reply:
I agree - it would be good to be able to apply this to a component, and then remove it, revealing content underneath when the data has loaded - probably tied to a selector/observable. Unless we decide that not to allow this by design - kind of relates my comment below :) https://github.com/Legal-and-General/canopy/pull/302#issuecomment-816532018